### ISSUE TITLE
allow-unwrap-in-tests does not work in unit test that are also gated by a feature flag

### ISSUE THREAD

**Issue by stormshield-gt** [rust-lang/rust-clippy#16369]
### Summary

I have a codebase where I want to prevent the use of unwrap outside of unit test.

For that I tried to use the lint `clippy::unwrap_unsed` with the configuration `allow-unwrap-in-tests=true`.

Some tests also depend on a feature flag being activated, i.e., `#[cfg(all(test, feature = "my-feature"))]`. For those there is a false positive for test helper functions.

### Reproducer

Code:

```rust
#[cfg(all(test, feature = "my-feature"))]
mod tests {
    fn helper() -> u32 {
        Some(42).unwrap()
    }

    #[test]
    fn test() {
        let _ = helper();
        let _ = Some(21).unwrap();
    }
}
```
Cargo.toml:

```toml
[package]
name = "playground"
version = "0.1.0"
edition = "2024"

[features]
my-feature = []

[lints.clippy]
unwrap_used = "deny"
```

Current output:

Warning that `unwrap()` is used in `helper()`

Desired output:

No warning

Note that the current workaround is to add another `cfg(test)`:
```rust
#[cfg(all(test, feature = "my-feature"))]
#[cfg(test)]
mod tests {
//..
}
```

### Version

```text
rustc 1.92.0 (ded5c06cf 2025-12-08)
binary: rustc
commit-hash: ded5c06cf21d2b93bffd5d884aa6e96934ee4234
commit-date: 2025-12-08
host: x86_64-unknown-linux-gnu
release: 1.92.0
LLVM version: 21.1.3
```
